apis/apiextensions/v1/composition_types.go

```diff
@@ -243,6 +250,73 @@ type Patch struct {
 	// Policy configures the specifics of patching behaviour.
 	// +optional
 	Policy *PatchPolicy `json:"policy,omitempty"`
+
+	// FromFn is a function whose output is patched instead of a reference to a
+	// field path
+	// +optional
+	FromFn *FromFn `json:"fromFn,omitempty"`
```

Comment:
I know it was me who suggested `fromFn` as a pattern in https://github.com/crossplane/crossplane/issues/1895, but I feel like this API needs some more thought. In particular, it's a little confusing that we would have "fromFn" (which takes no inputs) and "transform functions" that do take inputs. 🤔

Reply:
I got a little confused too. I saw transformers like `FromFn` but with the restriction of taking a `FromField` input, if remove that restriction `RandomString` can just be a transformer applied without input.